### PR TITLE
Add notes for starting a development environment

### LINKED ISSUE #309: Write documentation for developers
Focus on developing with docker scenario.

### PR DESCRIPTION
**Pull request by effigies**
Added a sub-heading to Docker, since most of the prep is just getting the Docker image to run at all. Is this the sort of thing you were looking for, @chrisfilo?

Closes #309.

### CODE REVIEW

**Comment by chrisfilo** [comment]
Yup this is great. Some comments:

1. Explain how to build the container.
2. Explain how to add python and other dependencies.
3. When you mount external sources I was under the impression that it only works if a egg-info file is present see: https://github.com/poldracklab/fmriprep/wiki/Mounting-niworkflows-for-development

**Comment by effigies** [comment]
Updated. Think maybe with the amount of content, should move to a new file. What do you think?

Comments on the content itself are welcome, of course.

**Comment by rwblair** [comment]
A dedicated contributors file for this documentation would be good. I think its common in many larger projects.

File: docs/installation.rst [hunk 1]
```diff
@@ -23,6 +23,111 @@ For example: ::
...
+    root@03e5df018c5e:~# cd ~/src/fmriprep/
+    root@03e5df018c5e:~/src/fmriprep# pip install -e .
+    root@03e5df018c5e:~# cd ~/src/niworkflows/
+    root@03e5df018c5e:~/src/niworkflows# pip install -e .
+    root@03e5df018c5e:~# cd ~/src/nipype/
+    root@03e5df018c5e:~/src/nipype# pip install -e .
+
+Adding dependencies
+```````````````````
+New dependencies to insert into the Docker image will either be Python
```
Comment (rwblair) [inline docs/installation.rst]: Maybe rewrite:
"New dependencies to be inserted into"

**Comment by effigies** [comment]
Ready for review.

File: docs/contributors.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+Contributing to FMRIPREP
+------------------------
+
+This document explains how to prepare a new development environment and
+update an existing environment, as necessary.
+
+Development in Docker is encouraged, for the sake of consistency and
+portability.
+By default, work should be built off of `poldracklab/fmriprep:latest
+<https://hub.docker.com/r/poldracklab/fmriprep/>`_ (see the
```
Comment (oesteban) [inline docs/contributors.rst]: This looks like a link (with that underscore) but seems to be broken.

File: docs/contributors.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+installation_ guide for the basic procedure for running).
```
Comment (oesteban) [inline docs/contributors.rst]: This looks like a link (with that underscore) but seems to be broken.

Comment (effigies) [inline docs/contributors.rst]: Building it locally (make -C docs html), it points to installation.html. Is there a better way to do this?

Comment (oesteban) [inline docs/contributors.rst]: Sorry, now I realize that I was looking at the github advanced diff... should be fine.

**Comment by chrisfilo** [comment]
Great work!

**Pull request merged**